requirements-dev.txt

```diff
@@ -1,8 +1,7 @@
 coverage
+flake8
 mock
 nose
-pycodestyle==2.4.0
-pyflakes==1.6.0
```

Comment:
Would prefer to leave these in here to pin versions (feel free to update the version to the current/latest.) 

We've run in to problems in the past where CI starts failing unrelated to changes in PRs due to updates the CI process grabbing a newer version than the codebase has been vetted by. (new checks added or existing updated.)

Reply:
I tend to pin runtime dependencies but allow testing tools, linters, formatters, etc. to float.  As new tests are added/improved, the testing gets more rigorous and more issues that should get fixed are fixed which improves code quality and resiliency.  It does not matter if the issue is related to a particular PR.  What matters is the code continues to be improved.

Take pyflakes for instance...  Here it is hardcoded so none of the [improved tests](https://github.com/PyCQA/pyflakes/blob/master/NEWS.rst) between 1.6 and 2.1.1 have been run on this codebase.  That is unfortunate because they could have improved code quality and resiliency.

Reply:
Understand that, but the decision was made to trade that off and to update the pinned versions from time to time.

Reply:
Also note that if you go back in time and check out a specific release tag or SHA and install requirements it'll fail to pass linting b/c that code/tag wasn't tested with versions of things that didn't exist at the time. Having them pinned keeps released tested and validated with the version in use at the time. 

I expect dependabot to be able to file PRs to update these versions as the come available in the hopefully not too distant future which should get things to the ideal set up.